src/ripple/shamap/SHAMap.h

```diff
@@ -177,10 +177,29 @@ class SHAMap
     // traverse functions
     const_iterator upper_bound(uint256 const& id) const;
 
-    void visitNodes (std::function<bool (SHAMapAbstractNode&)> const&) const;
-    void
-        visitLeaves(
-            std::function<void(std::shared_ptr<SHAMapItem const> const&)> const&) const;
+    /**  Visit every node in this SHAMap
+
+         @param function called with every node visited.
+         If function returns false, visitNodes exits.
+    */
+    void visitNodes (std::function<bool (
+        SHAMapAbstractNode&)> const& function) const;
+
+    /**  Visit every node in this SHAMap that
+         is not present in the specified SHAMap
+
+         @param function called with every node visited.
+         If function returns false, visitDifferences exits.
+    */
+    void visitDifferences(SHAMap const* have,
+        std::function<bool (SHAMapAbstractNode&)>) const;
```

Comment:
Why was `visitDifferences` made public?  I don't see it used outside of `SHAMap::getFetchPack`.